internal/authtoken/repository.go

```diff
@@ -58,34 +59,35 @@ func NewRepository(r db.Reader, w db.Writer, kms *kms.Kms, opt ...Option) (*Repo
 // contains the auth token value. The provided IAM User ID must be associated to the provided auth account id
 // or an error will be returned. All options are ignored.
 func (r *Repository) CreateAuthToken(ctx context.Context, withIamUser *iam.User, withAuthAccountId string, opt ...Option) (*AuthToken, error) {
+	const op = "authtoken.(Repository).CreateAuthToken"
 	if withIamUser == nil {
-		return nil, fmt.Errorf("create: auth token: no user: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, op, "missing user")
 	}
 	if withIamUser.GetPublicId() == "" {
-		return nil, fmt.Errorf("create: auth token: no user id: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, op, "missing user id")
 	}
 	if withAuthAccountId == "" {
-		return nil, fmt.Errorf("create: auth token: no auth account id: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.InvalidParameter, op, "missing auth account id")
 	}
 
 	at := allocAuthToken()
 	at.AuthAccountId = withAuthAccountId
-
 	id, err := newAuthTokenId()
 	if err != nil {
-		return nil, fmt.Errorf("create: auth token id: %w", err)
+		//Unsure on this one
+		return nil, errors.Wrap(err, op, errors.WithMsg("auth token id"))
 	}
 	at.PublicId = id
 
 	token, err := newAuthToken()
 	if err != nil {
-		return nil, fmt.Errorf("create: auth token value: %w", err)
+		return nil, errors.Wrap(err, op)
 	}
 	at.Token = token
 
 	databaseWrapper, err := r.kms.GetWrapper(ctx, withIamUser.GetScopeId(), kms.KeyPurposeDatabase)
 	if err != nil {
-		return nil, fmt.Errorf("create: unable to get database wrapper: %w", err)
+		return nil, errors.Wrap(err, op)
```

Comment:
Rather keep the `unable to get database wrapper` as it clarifies the purpose of the kms.GetWrapper call `kms.KeyPurposeDatabase`